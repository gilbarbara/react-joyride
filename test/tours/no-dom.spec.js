import React from 'react';

import Standard from '../__fixtures__/Standard';

jest.mock('exenv', () => ({
  canUseDOM: false,
}));

console.warn = jest.fn();

const mockCallback = jest.fn();
const props = {
  callback: mockCallback,
};

describe('Joyride > NO-DOM', () => {
  let wrapper;
  let joyride;

  beforeAll(() => {
    wrapper = mount(<Standard {...props} />, { attachTo: document.getElementById('react') });
    joyride = wrapper.find('Joyride').instance();
  });

  it('should have initiated the Joyride', () => {
    expect(joyride.state).toMatchSnapshot();
  });

  it('should not be able to start the tour', () => {
    wrapper.find('.hero__start').simulate('click');

    expect(wrapper.find('Joyride > div')).not.toExist();
  });
});
