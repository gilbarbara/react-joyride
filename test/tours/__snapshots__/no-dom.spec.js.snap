// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Joyride > NO-DOM should have initiated the Joyride 1`] = `
{
  "action": "init",
  "controlled": false,
  "index": 0,
  "lifecycle": "init",
  "size": 4,
  "status": "ready",
}
`;
